src/OrchardCore/OrchardCore.ReCaptcha.Core/Services/ReCaptchaClient.cs

```diff
@@ -42,7 +41,9 @@ public async Task<bool> VerifyAsync(string responseToken, string secretKey)
             });
             try
             {
-                var response = await _httpClient.PostAsync("siteverify", content);
+                var client = _httpClientFactory.CreateClient(Name);
```

Comment:
This is assuming that `IHttpClientFactory` is registered at the tenant level and not the host level so that each tenant for the same resolved name will get a different instance, because it has a custom configuration (Api Url).

But if we didn't set the base uri and have it instead in the POST call then we can even share the same client for all services in the tenant without base Uri.

Reply:
I think here the options are evaluated every time the new client is created. So it would use the current tenant's options.  We may have to remove the ReCaptchSettings from the singleton service too

Reply:
@sebastienros I did some cleanup. I think `ReCaptchaService` is okay to make it scoped. The `AddHttpClient` adds the client as transient.

Reply:
@sebastienros I did some cleanup. I think `ReCaptchaService` is okay to make it scoped. Since there is no abstractions here, I removed `ReCaptchaClient`. I think it's over engineering and adds unnecessary complexity. Now, we register `ReCaptchaService` as scoped service which should be good in this case. Also, we now use `IHttpClientFactory`.

Reply:
IHttpClientFactory, was it removed in the meantime?

Reply:
We still use it. We register HttpClient for the type, so injecting HttpClient in the constructor will return the same instance for that type